include/pisa/query/queries.hpp

```diff
@@ -92,12 +92,13 @@ struct Query {
 
 [[nodiscard]] std::function<void(const std::string)> compute_parse_query_function(
     std::vector<Query> &queries,
-    const std::optional<std::string> terms_file,
-    const std::optional<std::string> stopwords_filename,
-    const std::optional<std::string> stemmer_type) {
+    std::optional<std::string> const &terms_file,
+    std::optional<std::string> const &stopwords_filename,
+    std::optional<std::string> const &stemmer_type)
+{
     if (terms_file) {
-        auto term_processor = TermProcessor(terms_file, stopwords_filename, stemmer_type);
-        return [&](std::string const &query_line) {
```

Comment:
Sorry for this bug introduced after refactoring. I understand that it was caused because `term_processor` was destroyed after exiting of the function, is that correct?

In the other hand, would it be an advantage capturing only `&queries` instead `&`?

Reply:
It might be a good idea to capture things one by one, especially when capturing by reference. Nevertheless, we have to move term processor into a generalized capture because there's no capture-by-move operator, like `&` for references and `=` for copies. the alternative to what I did is:
```cpp
auto term_processor = TermProcessor(terms_file, stopwords_filename, stemmer_type);
return [&queries, term_processor = std::move(term_processor)](
           std::string const &query_line) {
    queries.push_back(parse_query_terms(query_line, term_processor));
};
```
Here, you explicitly move what is previously constructed.

Reply:
On another note, we should be using some static code analyzer in our CI to find these things. I'm reasonably sure, say, `clang-tidy` would find this.